src/blockchain/web3_extentions/lido_contracts.py

```diff
@@ -32,12 +45,25 @@ def _load_contracts(self):
             ContractFactoryClass=LidoContract,
         ))
 
-        self.deposit_security_module = cast(DepositSecurityModuleContract, self.w3.eth.contract(
-            address=self.lido_locator.deposit_security_module(),
-            ContractFactoryClass=DepositSecurityModuleContract,
-        ))
-
         self.staking_router = cast(StakingRouterContract, self.w3.eth.contract(
             address=self.lido_locator.staking_router(),
             ContractFactoryClass=StakingRouterContract,
         ))
+
+        dsm_address = self.lido_locator.deposit_security_module()
+
+        self.deposit_security_module = cast(DepositSecurityModuleContractV2, self.w3.eth.contract(
+            address=dsm_address,
+            ContractFactoryClass=DepositSecurityModuleContractV2,
+        ))
+
+        try:
+            self.deposit_security_module.version()
+        except ContractLogicError:
```

Comment:
If the issue wasn't caused by the absence of a method, but rather by another network error that resulted in a "ContractLogicError," how would we handle it?